flight/odlc_boundaries/search_path.py

```diff
@@ -0,0 +1,204 @@
+        For complete photographic coverage of the area, this should be equal to half the height
+        of the area the camera covers on the ground given the current altitude.
+
+    Returns
+    -------
+    generated_search_paths : [Tuple[float, float]]
+        A list of concentric search paths that cover the area of the polygon
+    """
+
+    # convert to shapely polygon for buffer operations
+    search_area_points: list[tuple[float, float]] = [
+        (point["utm_x"], point["utm_y"]) for point in d_search_area_boundary
+    ]
+    boundary_shape: Polygon = Polygon(search_area_points)
+
+    generated_search_paths: list[tuple[float, float]] = []
+    # shrink boundary by a fixed amount until the area it covers is 0
+    # add the smaller boundary to our list of search paths on each iteration
+    while boundary_shape.area > 0:
+        # print(boundary_shape.exterior.coords.xy)
+        generated_search_paths.append(
+            tuple(zip(*boundary_shape.exterior.coords.xy))  # pylint: disable=maybe-no-member
+        )
+        boundary_shape = boundary_shape.buffer(buffer_distance, single_sided=True)
+
+    return generated_search_paths
+
+
+async def run() -> None:
+    """
+    This function is just a driver to test the goto function and runs through the
+    entire waypoint section of the SUAS competition
+    """
+
+    waypoint: dict[str, list[float]] = {
+        "lats": [38.31451966813249, 38.31430872867596, 38.31461622313521],
+        "longs": [-76.54519982319357, -76.54397320409971, -76.54516993186949],
+        "Altitude": [85],
+    }
+
+    # create a drone object
+    drone: System = System()
+    await drone.connect(system_address="udp://:14540")
+
+    # initilize drone configurations
+    await drone.action.set_takeoff_altitude(12)
+    await drone.action.set_maximum_speed(20)
+
+    # connect to the drone
+    logging.info("Waiting for drone to connect...")
+    async for state in drone.core.connection_state():
+        if state.is_connected:
+            logging.info("Drone discovered!")
+            break
+
+    logging.info("Waiting for drone to have a global position estimate...")
+    async for health in drone.telemetry.health():
+        if health.is_global_position_ok:
+            logging.info("Global position estimate ok")
+            break
+
+    logging.info("-- Arming")
+    await drone.action.arm()
+
+    logging.info("-- Taking off")
+    await drone.action.takeoff()
+
+    # wait for drone to take off
+    await asyncio.sleep(10)
+
+    # move to each waypoint in mission
+    point: int
+    for point in range(3):
+        await move_to(
+            drone, waypoint["lats"][point], waypoint["longs"][point], waypoint["Altitude"][0]
+        )
+
+    # infinite loop till forced disconnect
+    while True:
+        await asyncio.sleep(1)
+
+    # Official Coordinates for Maryland
+
+
+if __name__ == "__main__":
+    data_search_area_boundary: list[dict[str, float]] = [
```

Comment:
The comment about this coordinates should probably be directly above the line where they are defined, so move the comment to above line 154.